File: java/src/main/java/algorithm/sorting/SelectionSort.java
package algorithm.sorting;

import java.util.Arrays;

public class SelectionSort {

  static void selectionSort(int[] arr) {
    for (int i = 0; i < arr.length; i++) {
      int minIdx = i;
      for (int j = i + 1; j < arr.length; j++) {
        if (arr[j] < arr[minIdx]) {
          minIdx = j;
        }
      }
      int temp = arr[minIdx];
      arr[minIdx] = arr[i];
      arr[i] = temp;
    }
  }

  public static void main(String[] args) {
    int[] input = {1, 3, 7, 4, 2};
    Arrays.stream(input).forEach(System.out::print);
    selectionSort(input);
    System.out.println();
    Arrays.stream(input).forEach(System.out::print);
  }
}

File: java/src/main/kotlin/algorithm/sorting/SelectionSort.kt
package kotlin.algorithm.sorting

fun selectionSort(arr: IntArray) {
    for (i in 0 .. arr.size-1) {
       var minIdx = i
       for (j in 0 .. arr.size-1) {
           if (arr[j] < arr[minIdx]) {
               minIdx = j
           }
       }
        var temp = arr[i]
        arr[i] = arr[minIdx]
        arr[minIdx] = temp
    }
}

fun main(){

}